src/clarity-angular/data/datagrid/providers/selection.ts

```diff
@@ -39,7 +39,7 @@ export class Selection {
             }
             let leftOver: any[];
             if (this._items.trackBy) {
-                const trackBy: TrackByFn = this._items.trackBy;
+                const trackBy: TrackByFunction<Function> = this._items.trackBy;
                 const updatedTracked: any[] = updatedItems.map((item, index) => trackBy(index, item));
                 leftOver = this.current.filter((selected, index) => {
                     return updatedTracked.indexOf(trackBy(index, selected)) > -1;
```

Comment:
I didn't realize this before, but this is wrong. The index of the item in `current` has nothing to do with the index of the item in the array of items itself. So any trackBy function that actually uses the index would break.

Reply:
However, logically using the index for trackby and server driven list wouldn't yield the results you want. So while it doesn't send the right index, the use of that index (regardless of the value) would not be reflective of the actual index (of a paginated server driven list). 

So what would be the 'right' index?